lib/redis_memo/memoizable/dependency.rb

```diff
@@ -50,9 +50,12 @@ def depends_on(dependency, **conditions)
   private
 
   def self.extract_from_relation(relation)
+    connection = ActiveRecord::Base.connection
+    unless connection.respond_to?(:dependency_of)
+      raise RedisMemo::WithoutMemoization, 'Caching active record queries is currently disabled on RedisMemo'
+    end
```

Comment:
curious how do we decided that when the kill switch is on, we want to check  if`connection.respond_to?(:dependency_of)` is true (is it because of the line 67 below?)
and are there other things we should consider besides checking the `dependency_of`?

Reply:
I noticed sentry errors for this when Donald/Michael were doing testing in staging related to the rails 6 upgrade:
```
undefined method `dependency_of' for #<ActiveRecord::ConnectionAdapters::PostgreSQLAdapter:0x0000560868d7d270>
```

This piece of code assumes that the `connection.exec_query` method is memoized, but that code doesn't get executed if we set the env var to disable cached select. Based on https://github.com/chanzuckerberg/redis-memo/blob/main/lib/redis_memo/memoize_method.rb#L9 , I think checking just `dependency_of` should be good enough.